store/store_test.go

```diff
@@ -127,23 +127,23 @@ func TestStoreGetSorted(t *testing.T) {
 	e, err := s.Get("/foo", true, true)
 	assert.Nil(t, err, "")
 	assert.Equal(t, e.EtcdIndex, eidx, "")
-	var yNodes NodeExterns
-	for _, node := range e.Node.Nodes {
-		switch node.Key {
-		case "/foo/x":
-		case "/foo/y":
-			yNodes = node.Nodes
-		case "/foo/z":
-		default:
-			t.Errorf("key = %s, not matched", node.Key)
+
+	sortedStrings := []string{"/foo/x", "/foo/y", "/foo/z"}
+	for i := range e.Node.Nodes {
+		node := e.Node.Nodes[i]
+		if node.Key != sortedStrings[i] {
+			t.Errorf("expect key = %s, got key = %s", sortedStrings[i], node.Key)
 		}
 	}
-	for _, node := range yNodes {
-		switch node.Key {
-		case "/foo/y/a":
-		case "/foo/y/b":
-		default:
-			t.Errorf("key = %s, not matched", node.Key)
+
+	e, err = s.Get("/foo/y", true, true)
```

Comment:
right, but instead of fetching "/foo/y" with `Get` I think it needs to be something like:
```go
var yNodes NodeExterns
for _, node := range e.Node.Nodes {
   if node.Key == "/foo/y" {
       yNodes = node.Nodes
       break
   }
}
sortedStrings = []string{"/foo/y/a", "/foo/y/b"}
for i := range yNodes {
    if yNodes[i].Key != sortedStrings[i] {
        t.Errorf(...)
    }
}
```

Otherwise, it's not testing the full tree from the response to "/foo".